public/app/features/variables/utils.ts

```diff
@@ -51,7 +51,7 @@ export const getSearchFilterScopedVar = (args: {
 
 export function containsVariable(...args: any[]) {
   const variableName = args[args.length - 1];
-  args[0] = isString(args[0]) ? args[0] : Object['values'](args[0]).join(' ');
+  args[0] = isString(args[0]) ? args[0] : JSON.stringify(args[0]);
```

Comment:
JSON stringify can blow up if the object is circular maybe use something like safeStringifyValye here https://github.com/grafana/grafana/blob/6dfa9b482351edc1da564e4e7917876f9b645d1c/public/app/core/utils/explore.ts#L202

Reply:
Thank you @hugohaggmark ! This should be fine now.
nb: I'll add test a bit later and squash the commits then.